File: server/models/AdminPaymentRequest.js
const mongoose = require("mongoose");

const AdminPaymentRequestSchema = new mongoose.Schema({
  amount: { type: Number, required: true },
  // You may add other admin-only fields (like a title, description, or expiry)
  status: { type: String, default: "open" }, // For example: open, used, etc.
  createdAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model("AdminPaymentRequest", AdminPaymentRequestSchema);

File: lib/models/admin_payment_request.dart
class AdminPaymentRequest {
  final String id;
  final double amount;
  final String status;
  final DateTime createdAt;

  AdminPaymentRequest({
    required this.id,
    required this.amount,
    required this.status,
    required this.createdAt,
  });

  factory AdminPaymentRequest.fromJson(Map<String, dynamic> json) {
    return AdminPaymentRequest(
      id: json['_id'] ?? "",
      amount: (json['amount'] as num).toDouble(),
      status: json['status'] ?? "open",
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}
